src/main/java/com/comcast/zucchini/AbstractZucchiniTest.java

```diff
@@ -132,6 +147,8 @@ private boolean envSerialized() {
      * @param contexts The list of contexts that will be run.
      */
     public void runParallel(List<TestContext> contexts) {
+        this.canBarrierInTest = this.canBarrier;
```

Comment:
instead of setting this property from both runParallel and runSerial, why not call it once from run?